test/distributed/org/apache/cassandra/distributed/test/ShortReadProtectionTest.java

```diff
@@ -66,51 +329,192 @@ public void testGroupBySRPRegularRow() throws Throwable
-                       asList("SELECT * FROM %s LIMIT 1",
-                              "SELECT * FROM %s LIMIT 10",
-                              "SELECT * FROM %s GROUP BY pk LIMIT 1",
-                              "SELECT * FROM %s GROUP BY pk LIMIT 10",
-                              "SELECT * FROM %s GROUP BY pk, ck LIMIT 1",
-                              "SELECT * FROM %s GROUP BY pk, ck LIMIT 10"));
-    }
-
-    private void testGroupBySRP(String createTable,
-                                List<String> node1Queries,
-                                List<String> node2Queries,
-                                List<String> coordinatorQueries) throws Throwable
-    {
-        try (Cluster cluster = init(Cluster.build()
-                                           .withNodes(2)
-                                           .withConfig(config -> config.set("hinted_handoff_enabled", false))
-                                           .start()))
+    public void testGroupByStaticRow()
+    {
+        Assume.assumeFalse(paging); // paging fails due to CASSANDRA-16307
+        tester.createTable("CREATE TABLE %s (pk int, ck int, s int static, PRIMARY KEY (pk, ck))")
+              .onlyNode1("INSERT INTO %s (pk, s) VALUES (1, 1) USING TIMESTAMP 0",
+                         "INSERT INTO %s (pk, s) VALUES (0, null)",
+                         "INSERT INTO %s (pk, s) VALUES (2, 2) USING TIMESTAMP 0")
+              .onlyNode2("INSERT INTO %s (pk, s) VALUES (1, null)",
+                         "INSERT INTO %s (pk, s) VALUES (0, 0) USING TIMESTAMP 0",
+                         "INSERT INTO %s (pk, s) VALUES (2, null)")
+              .assertRows("SELECT * FROM %s LIMIT 1")
+              .assertRows("SELECT * FROM %s LIMIT 10")
+              .assertRows("SELECT * FROM %s GROUP BY pk LIMIT 1")
+              .assertRows("SELECT * FROM %s GROUP BY pk LIMIT 10")
+              .assertRows("SELECT * FROM %s GROUP BY pk, ck LIMIT 1")
+              .assertRows("SELECT * FROM %s GROUP BY pk, ck LIMIT 10");
+    }
+
+    /**
+     * See CASSANDRA-13911.
+     * <p>
+     * Replaces Python dtest {@code consistency_test.py:TestConsistency.test_13911()}.
+     */
+    @Test
+    public void test13911()
+    {
+        tester.createTable("CREATE TABLE %s (pk int, ck int, PRIMARY KEY (pk, ck))")
+              .onlyNode1("INSERT INTO %s (pk, ck) VALUES (0, 0)")
+              .onlyNode2("DELETE FROM %s WHERE pk = 0 AND ck IN (1, 2)")
+              .assertRows("SELECT DISTINCT pk FROM %s", row(0));
+    }
+
+    /**
+     * A regression test to prove that we can no longer rely on {@code !singleResultCounter.isDoneForPartition()} to
+     * abort single partition SRP early if a per partition limit is set.
+     * <p>
+     * See CASSANDRA-13911.
+     * <p>
+     * Replaces Python dtest {@code consistency_test.py:TestConsistency.test_13911_rows_srp()}.
+     */
+    @Test
+    public void test13911rows()
```

Comment:
nit: Maybe we can give these better names now, especially since the Jira is already in the JavaDoc above.

Reply:
Yes, I used more descriptive name for some other tests but not for this one due to a lack of imagination. [Here](https://github.com/apache/cassandra/pull/801/commits/057756b50ed630bf6ce9ddd0f51e8ade58b172f8) I have used `testSkipEarlyTermination*` instead of `test13911*`, wdyt?

Reply:
Those names work for me.